src/Storages/Kafka/StorageKafka.cpp

```diff
@@ -244,69 +244,130 @@ namespace
+                //         </kafka_topic>
+                //     </kafka>
+                // Advantages: The period restriction no longer applies (e.g. <name>sports.football</name> will work), everything
+                // Kafka-related is below <kafka>.
+                for (const auto & topic : topics)
                 {
-                    if (inner_tag == CONFIG_NAME_TAG)
-                        continue; // ignore <name>
+                    /// Read topic name between <name>...</name>
+                    const String kafka_topic_path = config_prefix + "." + tag;
+                    const String kafpa_topic_name_path = kafka_topic_path + "." + CONFIG_NAME_TAG;
 
-                    /// "log_level" has valid underscore, the remaining librdkafka setting use dot.separated.format which isn't acceptable for XML.
-                    /// See https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md
-                    const String setting_path = kafka_topic_path + "." + inner_tag;
-                    const String setting_value = config.getString(setting_path);
+                    const String topic_name = config.getString(kafpa_topic_name_path);
 
-                    const String setting_name_in_kafka_config = (inner_tag == "log_level") ? inner_tag : boost::replace_all_copy(inner_tag, "_", ".");
-                    kafka_config.set(setting_name_in_kafka_config, setting_value);
+                    if (topic_name != topic)
+                        continue;
+                    loadTopicConfig(kafka_config, config, kafka_topic_path);
                 }
+                continue;
             }
+            if (tag.starts_with(CONFIG_KAFKA_TAG)) /// skip legacy configuration per topic e.g. <kafka_TOPIC_NAME>.
+                continue;
+            // Update configuration from the configuration. Example:
+            //     <kafka>
+            //         <retry_backoff_ms>250</retry_backoff_ms>
+            //         <fetch_min_bytes>100000</fetch_min_bytes>
+            //     </kafka>
+            loadConfigProperty(kafka_config, config, config_prefix, tag);
+        }
+    }
+
+    void loadTopicLegacyConfig(cppkafka::Configuration & kafka_config, const Poco::Util::AbstractConfiguration & config, const String & prefix, const Names & topics)
+    {
+        // Update consumer topic-specific configuration (legacy syntax, retained for compatibility). Example with topic "football":
+        //     <kafka_football>
+        //         <retry_backoff_ms>250</retry_backoff_ms>
+        //         <fetch_min_bytes>100000</fetch_min_bytes>
+        //     </kafka_football>
+        // The legacy syntax has the problem that periods in topic names (e.g. "sports.football") are not supported because the Poco
+        // configuration framework hierarchy is based on periods as level separators. Besides that, per-topic tags at the same level
+        // as <kafka> are ugly.
+        String topic_path = prefix;
```

Comment:
The stuff on l. 341-343 is new and I don't get why it is needed.

Reply:
Also, removed.